activerecord/test/cases/associations/inverse_associations_test.rb

```diff
@@ -670,6 +670,21 @@ def test_inverse_works_when_the_association_self_references_the_same_object
     assert_equal comment.body, comment.children.first.parent.body
   end
 
+  def test_inverse_works_when_the_association_self_references_the_same_object_and_uses_non_id_foreign_key
+    post = Post.create!(token: "A-123", title: "My Post", body: "Post Body")
+    parent = UnconventionalComment.create!(uuid: "d6d98b88", post: post, label: :default, body: "New Parent Comment")
+    UnconventionalComment.create!(comment_uuid: parent.uuid, post: post, label: :child, body: "New Child Comment")
+
+    assert_equal parent.object_id, parent.children.first.parent.object_id
```

Comment:
it looks like your example is missing the call to `inverse_of:` for this test to work. 

> so im curious what happens if there is a non custom primary_key but custom scope situation, and how it interacts with the custom primary_key

I am not sure I am following this, but if I parsed this correctly 

> if there is a non custom primary_key but custom scope situation,

This works with this branch. Adding the scope you have still makes the test script pass

> and how it interacts with the custom primary_key

This is what I don't understand, what will interact with a custom primary key since you mentioned right before this that there is a non custom primary key

Reply:
Got it - it looks like adding a custom scope makes it so that any automatic inversing no longer works, it's not a behavior that I was aware of(nor seems to be documented). Adding an explicit inverse_of works.